corehq/apps/app_manager/models.py

```diff
@@ -4557,7 +4556,6 @@ def save(self, response_json=None, increment_version=None, **params):
         get_apps_in_domain.clear(self.domain, True)
         get_apps_in_domain.clear(self.domain, False)
         get_mobile_ucr_count.clear(self.domain)
-        clear_caches_case_data_cleaning(self.domain)
```

Comment:
Moving this to an async task means the cache will not be cleared until after save, where this was previously done before save. In either case there is a race condition. It's not clear which is better, but just noting that such race conditions can cause hard-to-find bugs.

It's also notable that moving this one to after save means the race conditions are split since some caches are still being cleared before.

Reply:
There is definitely a race condition. The question is whether it matters in practice. I have not done the research to know, and posed the question in hopes that you would.